Explorer/Assets/DCL/UI/GenericContextMenu/GenericContextMenuController.cs

```diff
@@ -89,33 +103,73 @@ protected override void OnViewShow()
             inputData.ActionOnShow?.Invoke();
         }
 
-        private void ConfigureContextMenu()
+        private void ConfigureContextMenu(ControlsContainerView container, GenericContextMenuParameter.GenericContextMenu contextMenuConfig, Vector2 anchorPosition, Rect? overlapRect)
         {
             float totalHeight = 0;
 
-            for (var i = 0; i < inputData.Config.contextMenuSettings.Count; i++)
+            for (var i = 0; i < contextMenuConfig.contextMenuSettings.Count; i++)
             {
-                GenericContextMenuElement config = inputData.Config.contextMenuSettings[i];
+                GenericContextMenuElement config = contextMenuConfig.contextMenuSettings[i];
 
                 if (!config.Enabled) continue;
 
-                GenericContextMenuComponentBase component = controlsPoolManager.GetContextMenuComponent(config.setting, i);
+                GenericContextMenuComponentBase component = controlsPoolManager.GetContextMenuComponent(config.setting, i, container.transform);
+
+                if (config.setting is SubMenuContextMenuButtonSettings subMenuButtonSettings && component is GenericContextMenuSubMenuButtonView subMenuButtonView)
```

Comment:
It's a bad design because it permits having desynchronization between the settings type and the view type which in reality is not true.
It's connected to how `GetContextMenuComponent`  method is designed.
I don't have a ready solution off the top of my head, though.